### PR TITLE
write out native ids and spectrum references

### PR DESCRIPTION
**Pull request by timosachsenberg**
add references to original data and native ids

### CODE REVIEW

**Comment by timosachsenberg** [comment]
might fix https://github.com/OpenMS/OpenMS/issues/4834

File: src/topp/OMSSAAdapter.cpp [hunk 1]
```diff
@@ -1018,6 +1019,33 @@ class TOPPOMSSAAdapter :
...
+    // reannotate file oriin and native ids
+    MSExperiment exp;
+    MzMLFile().load(inputfile_name, exp);
+    protein_identification.setPrimaryMSRunPath({inputfile_name}, exp);
+
+    // add RT and precursor m/z to the peptide IDs (look them up in the spectra):
+    SpectrumLookup lookup;
+    lookup.readSpectra(exp);
+
+    for (auto& p : peptide_ids)
```
Comment (cbielow) [inline src/topp/OMSSAAdapter.cpp:1031]: this for loop can be refactored into a member of SpectrumLookup?!
I think there are at least 2 copies of this code in place, which we could get rid of.
I can do that

Comment (timosachsenberg) [reply src/topp/OMSSAAdapter.cpp:1031]: this would be nice! thanks

**Comment by timosachsenberg** [comment]
please don't merge... found an issue

**Comment by jpfeuffer** [comment]
Looks good. Merge if you are ready.

**Pull request merged**